src/rdposix.h

```diff
@@ -108,12 +126,45 @@
  */
 static RD_INLINE RD_UNUSED
 void rd_usleep (int usec, rd_atomic32_t *terminate) {
+#ifdef __370__
+    int left = usec / 1000000;
+
+    while(usec > 0) {
+        unsigned long long stck_start;
+        unsigned long long stck_end;
+
+        __stckf(&stck_start);
+	/* if sleep return non 0 it was interupted
+	 * the return is teh number of second it had
+	 * _not_ slept
+	 */
+	if(left)
+	{
```

Comment:
urgh.. _that_ is begging for bugs. I mean the place of the braces, ok that is a taste issue. but the no braces for single statement is asking of Mariner 1 like bugs. sigh.